sdk/storage/Azure.Storage.Files/src/Models/FileSmbProperties.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Globalization;
+using System.Text;
+
+namespace Azure.Storage.Files.Models
+{
+    /// <summary>
+    /// The SMB properties for a file.
+    /// </summary>
+    public struct FileSmbProperties : IEquatable<FileSmbProperties>
+    {
+        /// <summary>
+        /// The file system attributes for this file.
+        /// </summary>
+        public NtfsAttributes? FileAttributes { get; set; }
```

Comment:
Because "None" is actually a value you can pass to the service.  If NtfsAttributes is not nullable, we would override the the File's NTFS attributes even if the user wished to maintain them.

Reply:
I'm not sure this would pass the Arch Board review.  Can we add something like `Preserve` to the enum as the default value so users don't accidentally get things wrong by using `None` when they really want `null`?  Or other thoughts about how to make the important difference less subtle?

Reply:
The user shouldn't have to require these fields if they don't want to change them.  These parameters should have never been required on the service in the first place.

I've changed FileNtfsAttributes to be a struct with bools, similar to FileSasPermissions to address this ambiguity.

Reply:
I think that works.